### ISSUE TITLE
Property excludeScope in JavadocMethod check does not work correctly

### ISSUE THREAD

**Issue by damianszczepanik** [checkstyle/checkstyle#2205]
Based on this http://checkstyle.sourceforge.net/config_javadoc.html#JavadocMethod documentation I understand that by default it checks all scopes except private. In other words this is relevant to: 

```xml
        <module name="JavadocMethod">
            <property name="excludeScope" value="private"/>
        </module>
```

However I found in damianszczepanik/silencio/pull/42 that above comments generates 0 errors while following one
```xml
        <module name="JavadocMethod">
        </module>
```
complains that https://github.com/damianszczepanik/silencio/blob/b65764da81d118f40f19ee5e640fb2540bbba7c6/src/main/java/pl/szczepanik/silencio/diagnostics/ProcessorSmokeChecker.java#L45 has violation ``ProcessorSmokeChecker.java:45:5: Missing a Javadoc comment``

Is that correct that excluding ``private`` from checker generates errors for ``public``?

**Comment by mkordas** [comment]
The problem is in [JavadocMethodCheck.java#L465-L468](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java#L465-L468).
I'm on it.

**Comment by mkordas** [comment]
There is also another problem with `excludeScope` property. If it is set to `protected` as in [com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheckTest#testExcludeScope](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java#L349), it still reports missing Javadoc on `protected` members. Test proves that incorrect behaviour e.g. by 
```java
"46:9: " + getCheckMessage(MSG_JAVADOC_MISSING),
```
when this line is just
```java
protected void foo2() {}
```
